### ISSUE TITLE
[@typescript-eslint/no-unsafe-member-access] doesn't work with optional chaining 

### ISSUE THREAD

**Issue by karlismelderis** [typescript-eslint/typescript-eslint#2220]
**Repro**
```TS
let expressRoute = req.route?.path
let expressRoute = req?.route?.path
```

**Expected Result**

I expect eslint to respect that if it's optional chaining then don't bother that I'm reading from "any"

**Actual Result**

getting this error: `ESLint: Unsafe member access .path on an any value.`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.1` |
| `@typescript-eslint/parser`        | `3.0.1` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.0.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.14.5` |

**Comment by karlismelderis** [comment]
Even if I have object properly typed this is still giving errors:
`retryCount: error?.config?.['axios-retry']?.retryCount,`

as soon as there is ['some-prop'] `@typescript-eslint/no-unsafe-member-access`, `@typescript-eslint/no-unsafe-assignment` rules freak out.

**Comment by bradzacher** [comment]
It doesn't matter if you're using optional chaining or not - it's not a safe member access.

You should avoid ever using `any`. If it's a library type, consider asserting it to `unknown` before doing checks on it to validate the structure of the object.

**Comment by karlismelderis** [comment]
:) thank you for clarification
